ffi/diplomat/src/script.rs

```diff
@@ -32,20 +38,68 @@ pub mod ffi {
 
         /// Get the Script property value for a code point
         #[diplomat::rust_link(
-            icu::properties::script::ScriptWithExtensions::get_script_val,
+            icu::properties::script::ScriptWithExtensionsBorrowed::get_script_val,
             FnInStruct
         )]
         pub fn get_script_val(&self, code_point: u32) -> u16 {
-            self.0.get().data.get_script_val(code_point).0
+            self.0.as_borrowed().get_script_val(code_point).0
+        }
+
+        /// Check if the Script_Extensions property of the given code point covers the given script
+        #[diplomat::rust_link(
+            icu::properties::script::ScriptWithExtensionsBorrowed::has_script,
+            FnInStruct
+        )]
+        pub fn has_script(&self, code_point: u32, script: u16) -> bool {
+            self.0.as_borrowed().has_script(code_point, Script(script))
+        }
+
+        /// Borrow this object for a slightly faster variant with more operations
+        #[diplomat::rust_link(
+            icu::properties::script::ScriptWithExtensions::as_borrowed,
+            FnInStruct
+        )]
+        pub fn as_borrowed<'a>(&'a self) -> Box<ICU4XScriptWithExtensionsBorrowed<'a>> {
+            Box::new(ICU4XScriptWithExtensionsBorrowed(self.0.as_borrowed()))
         }
+    }
 
+    impl<'a> ICU4XScriptWithExtensionsBorrowed<'a> {
```

Comment:
Thought: We don't expose `CodePointSetDataBorrowed` due to simplicity and I think the assumption that the cost of crossing FFI overshadows the cost of the yoke. Do we need `ScriptWithExtensionsBorrowed` over FFI then?

Reply:
We have to for the scx set stuff to work, from yesterday's discussion. That's part of why I wanted to go back and redesign this.